lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -473,6 +482,47 @@ defmodule Screens.V2.CandidateGenerator.Dup.Departures do
       MapSet.disjoint?(not_informed, informed_stop_ids)
   end
 
+  defp overnight_time_period?(todays_schedules, now) do
+    first_last_schedule_times =
+      todays_schedules
+      |> Enum.map(&{&1.route.id, &1.direction_id})
+      |> Enum.uniq()
+      |> Enum.reduce({nil, nil}, fn {route_id, direction_id}, acc ->
+        first_schedule_today =
+          Enum.find(
+            todays_schedules,
+            &(&1.route.id == route_id and &1.direction_id == direction_id)
+          )
+
+        last_schedule_today =
+          Enum.find(
+            Enum.reverse(todays_schedules),
+            &(&1.route.id == route_id and &1.direction_id == direction_id)
+          )
```

Comment:
There seems to be a lot of overlap between this function and `get_overnight_schedules_for_section`, with some code duplicated outright. Could these be consolidated? Maybe `overnight_time_period?` could take as an input the values returned from that function?